### DISCUSSION TITLE
For a gas meter measuring in CCF, "Volume per half rotation" should be 0.00125

### DISCUSSION THREAD

**Discussion by 42Network** [tronikos/esphome-magnetometer-water-gas-meter#10] [General]
Hello. This is such an awesome project! I was very easily able to get some readings from my basic diaphragm-based gas meter. But I was very surprised to see that HA was telling me I was going to be getting a gas bill for ~$300 for just a few days of usage in the summer! 

Your notes say that 0.125 is the correct value for a gas meter for "Volume per half rotation" but I guess that assumes I told your integration to measure in ft^3 (cubic feet). A CCF is 100 cubit feet. So setting the value to 0.00125 makes the cost calculation in HA close to reality. 

1. Is my assumption (and math) correct to set "Volume per half rotation" to 0.00125 for measuring gas in CCF?

Two other questions:

2. What is the best way to clean out the incorrect historical data from HA, including the long-term statistics, from when my "Volume per half rotation" was incorrect? It seems the ESP32 itself has a persistent local variable for the total CCF. If I just delete and re-add the ESP32 device, HA seems to still know about the historical values. I played around with the "Purge entity" statistics service call but I'm not sure what needs to be done. I also see you have a service call for making the CCF number match my bill. Should I use that?
3. Mechanically, can you comment on where the 0.125 comes from? This is the reciprocal of 1 / 8, which implies something inside the meter is "rotating" 8 times to measure out one cubic foot of gas. There are actually 2 diaphragms and 4 chambers. I have placed the magnetometer on the backside of my meter. Is it detecting the motion of only the closest of the two diaphragms?

Here is a Grafana graph I made while my hot water heater was running. It is rated at 40k BTUh, which means my "One rotation per half CF" test dial on my gas meter outside should turn 360 degrees in roughly 48 seconds. That times 2 is 96 seconds, and indeed, the Grafana graph shows 8 steps in the rotation count during this time.
 
![image](https://github.com/tronikos/esphome-magnetometer-water-gas-meter/assets/35043496/6665d9de-13e1-4830-b2ab-d80282fae01f)

 Thanks for a great HA solution!

**Comment by tronikos** [comment]
1. Correct. I updated the README and yaml to clarify.
2. I think in developer tools > statistics you can see orphaned statistics to remove them. Yes you might have to call the esphome exposed service too.
3. No idea.
